tools/migration/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"runtime"
+
+	"github.com/filecoin-project/go-filecoin/tools/migration/cmd"
+	"github.com/filecoin-project/go-filecoin/util/version"
+)
+
+func main() { // nolint: deadcode
+	if len(os.Args) < 2 {
+		showUsageAndExit(1)
+	}
+
+	if !version.Check(runtime.Version()) {
+		exitErr(fmt.Sprintf("Invalid go version: %s", runtime.Version()))
+	}
```

Comment:
Why do we need this?

Reply:
This is done in build/main.go and I pulled it from there because I assumed we want to make sure we are running the version of go required by go-filecoin.

Reply:
left in the new PR